Update limitations of SQLite (#1026)
cap-js/docs: java/cqn-services/persistence-services.md

CAP does support most of the major features on SQLite, although there are a few shortcomings that are listed here:

1. SQLite has only limited support for concurrent database access. You're advised to limit the connection pool to *1* as shown above (parameter `maximum-pool-size: 1`), which effectively serializes all database transactions.

> 2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity locale dependent.

Reviewer: ```suggestion
2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity depending on the locale.
```

Reviewer 2: ```suggestion
2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity depend on the locale.
```

I don't think `depending` fits here, or does it? :)

Reviewer: What do we want say? Maybe rephrasing helps:

_In the future, we might provide an option to enable case-sensitivity in search according to the users locale._

Is that correct?

Reviewer 2: Yes, I think so. :)

Author: > In the future, we might provide an option to enable case-sensitivity in search according to the users locale.

I am wondering what the benefit of this information is. I actually think we can just delete this sentence.

> 2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite. In the future, we might provide an option to make the case-sensitivity depend on the locale.

Reviewer: ```suggestion
2. The predicate function `contains` is supported. However, the search for characters in the word or phrase is case-insensitive in SQLite.
```

3. SQLite doesn't support [pessimistic locking](../working-with-cql/query-execution#pessimistic-locking).

> 4. Streaming of large object data isn't supported by SQLite. Hence, when reading or writing data of type `cds.LargeString` and `cds.LargeBinary` as a stream the framework temporarily materializes the content. Thus, storing large objects on SQLite can impact the performance.

Reviewer: ```suggestion
4. Streaming of large object data isn't supported by SQLite. Hence, when reading or writing data of type `cds.LargeString` and `cds.LargeBinary` as a stream, the framework temporarily materializes the content. Thus, storing large objects on SQLite can impact the performance.
```

5. Sorting of character-based columns is never locale-specific but if any locale is specified in the context of a query then case insensitive sorting is performed.

> 6. Views in SQLite are read-only. However, the CAP Java SDK supports some views to be updatable as described [here](../working-with-cql/query-execution#updatable-views).

Reviewer: ```suggestion
6. Views in SQLite are read-only. However, the CAP Java SDK supports some views to be updatable as described in [Updatable Views](../working-with-cql/query-execution#updatable-views).
```
